scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1895,7 +1895,6 @@ def run(self):
                 except queue.Empty:
                     break
                 else:
-                    inst.metrics.update(self.instances[inst.name].metrics)
```

Comment:
if the test plan has some former metrics inside, then this could be a issue. how about
```
if self.instances[inst.name].metrics:
    inst.metrics.update(self.instances[inst.name].metrics)
```

Reply:
Currently, there aren't many metrics, so we can simply use the metrics from the `done_queue` (from the runner). However, if someone adds static metrics (added before the processing pipeline), those metrics will be overwritten.
You can try this approach:
```
                    self.instances[inst.name].metrics.update(inst.metrics)
                    inst.metrics = self.instances[inst.name].metrics
                    inst.metrics["handler_time"] = inst.execution_time
                    self.instances[inst.name] = inst
```
First, update the metrics in self.instances with the updated metrics from the runner, then copy those combined metrics back to the runner instance.

Reply:
Ok, I see that it is also not needed. Metrics that are originally in self.instance, are also in istance from done_queue, because were passed to the runner. Updating metrics ere does not have any sense.